File: 2015/12/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"log"
)

func sumJSON(jsonObject interface{}, ignoreRed bool) float64 {
	var sum float64
	switch val := jsonObject.(type) {
	case map[string]interface{}:
		for _, v := range val {
			if ignoreRed && v == "red" {
				return 0
			}
			sum += sumJSON(v, ignoreRed)
		}
	case []interface{}:
		for _, v := range val {
			sum += sumJSON(v, ignoreRed)
		}
	case float64:
		sum += val
	}
	return sum
}

func main() {
	var jsonString []byte
	var jsonObject interface{}
	fmt.Scanln(&jsonString)
	if err := json.Unmarshal(jsonString, &jsonObject); err != nil {
		log.Fatal(err)
	}
	sum := int(sumJSON(jsonObject, false))
	fmt.Println("Part One:", sum)
	sum = int(sumJSON(jsonObject, true))
	fmt.Println("Part Two:", sum)
}

File: 2015/12/solution.py
import json
from typing import Any


def add_up_nums(d: Any, ignore_red=False) -> int:
    if isinstance(d, int):
        return d
    if isinstance(d, list):
        return sum(add_up_nums(el, ignore_red=ignore_red) for el in d)
    if isinstance(d, dict):
        if ignore_red and "red" in d.values():
            return 0
        return sum(add_up_nums(val, ignore_red=ignore_red) for val in d.values())
    return 0


d = json.loads(input())
print("Part One:", add_up_nums(d))
print("Part Two:", add_up_nums(d, ignore_red=True))
